scripts/build/termux_step_setup_variables.sh

```diff
@@ -187,6 +187,9 @@ termux_step_setup_variables() {
 	TERMUX_PYTHON_HOME=$TERMUX_PREFIX/lib/python${TERMUX_PYTHON_VERSION} # location of python libraries
 	TERMUX_PKG_MESON_NATIVE=false
 	TERMUX_PKG_CMAKE_CROSSCOMPILING=true
+	TERMUX_PKG_ONLY_INSTALLING=false # when set to true you indicate that the package can only be installed
```

Comment:
To put it bluntly, a package with this variable set to `true` will not be added to the package's list of dependencies (by `buildorder.py`) if those dependencies are planed to be compiled and not installed.

I added this variable because at that time the dependency compilation scheme did not allow compiling complex packages that required many dependencies (one example of such a package is [gcc-libs-glibc](https://github.com/termux-pacman/glibc-packages/blob/main/gpkg/gcc-libs/build.sh#L14)). But now I have changed the dependency compilation scheme. Also, `build-package.sh` can work with cyclic dependencies, which also simplifies the compilation of a complex package as a dependency.

Reply:
Reopening. If we have to add new variables - and we should be trying as hard as possible not to add new variables, because there are *tons* of them - there needs to be a comment on this good enough that if someone sees `TERMUX_PKG_ONLY_INSTALLING` in a `build.sh` and looks here, they won't need to ask you or anyone else what it means. The comment you left just says exactly what's in its name. The response in this review isn't enough - the answer has to be in the code.

I also don't know what "compiled and not installed" means, so please think carefully about what'll make sense to a maintainer who's never seen this before. I will gladly help with the phrasing once you make it clear what it does and why we need to have it.